session/session.go

```diff
@@ -2916,6 +2917,24 @@ func (s *session) PrepareTSFuture(ctx context.Context) {
 		// In case the latest tso is misused.
 		return
 	}
+	if s.GetSessionVars().StmtCtx.WeakConsistency {
+		if s.txn.Valid() {
+			return
+		}
+		startTs := uint64(math.MaxUint64)
```

Comment:
Is it more likely to meet inconsistence and schema change error when using `MaxUint64` as start ts instead of a timestamp got from PD?

Reply:
A timestamp from PD prevents reading the data whose `commitTS > startTS`, there will be less consistency error than using `MaxUint64`.
The schema change error comes when the latest data is skipped by RC, so I think PD's timestamp is not helpful for this case.

Reply:
Restore the fetching timestamp part, RC reads will also request a timestamp from PD, and remove the invariant that RC read should use `MaxUint64`.